content/doc/book/pipeline/jenkinsfile.adoc

```diff
@@ -889,6 +853,177 @@ within a single `withCredentials( ... ) { ... }` step by doing the following:
+Using single-quotes instead of double-quotes when referencing these sensitive environment variables prevents this type of leaking.
+
+```
+pipeline {
+    agent any
+    environment {
+        EXAMPLE_CREDS = credentials('example-credentials-id')
+    }
+    stages {
+        stage('Example') {
+            steps {
+                /* CORRECT */
+                sh('curl -u $EXAMPLE_CREDS_USR:$EXAMPLE_CREDS_PSW https://example.com/')
+            }
+        }
+    }
+}
+```
+
+
+==== Injection via interpolation
+
+[WARNING]
+======
+Groovy string interpolation can inject rogue commands into command interpreters via special characters.
+======
+
+Another note of caution. Using Groovy string interpolation for user-controlled variables with steps that pass their arguments to command interpreters such as the `sh`, `bat`, `powershell`, or `pwsh` steps can result in problems analogous to SQL injection.
+This occurs when a user-controller variable (generally an environment variable, usually a parameter passed to the build) that contains special characters (e.g. `' " / \ $ & % ^ > < | ;`) is passed to the `sh`, `bat`, `powershell`, or `pwsh` steps using Groovy interpolation.
+For a simple example:
+
+```
+pipeline {
+  agent any
+  parameters {
+    password(name: 'EXAMPLE_SECRET', defaultValue: 'sec>ret', description: 'Enter a password')
+  }
+  stages {
+    stage('Example') {
+      steps {
+        sh "echo $EXAMPLE_SECRET"
+      }
+    }
+  }
+}
+```
+In this example, the argument to the `sh` step is evaluated by Groovy and `SECRET_VALUE` is interpolated directly into the argument as if `sh('echo sec>ret')` has been written in the Pipeline.
+When this is processed on the agent, rather than echoing the value `sec>ret`, it will create a file called `ret` containing the value `sec`.
+Any user able to control a variable interpolated by such a step would be able to make the `sh` step run arbitrary code on the agent.
+To avoid this problem, make sure arguments to steps such as `sh` or `bat` that reference parameters or other user-controller environment variables use single quotes to avoid Groovy interpolation.
+
+```
+pipeline {
+  agent any
+  parameters {
+    password(name: 'EXAMPLE_SECRET', defaultValue: 'sec>ret', description: 'Enter a password')
+  }
+  stages {
+    stage('Example') {
+      steps {
+        sh 'echo $EXAMPLE_SECRET'
+      }
+    }
+  }
+}
+```
+Credential mangling is another issue that can occur when credentials that contain special characters are passed to a step using Groovy interpolation.
+When the credential value is mangled, it is no longer valid and will no longer be masked in the console log.
+
+.NOTE: When using password parameters, the Masked Passwords plugin should be installed to mask sensitive parameters from the console
```

Comment:
I would not even mention this. Mask Passwords has no active maintainer and in general credentials should be used instead to avoid this problem, and this page already discusses credentials in detail.

```suggestion
```

Reply:
removed in 14207f3170d53d803556c01f4b56468ddc587615